samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/evaluation/TotalScoresBean.java

```diff
@@ -220,6 +232,53 @@ protected void init() {
 		
 		agents = newAgents;	
 	}
+
+  public boolean isOneSelectionType() {
+    if (this.getPublishedAssessment() != null) {
+      for (Object itemObject : this.getPublishedAssessment().getDefaultSection().getItemArray()) {
+        PublishedItemData item = (PublishedItemData) itemObject;
+        if (!(item.getTypeId().equals(TypeIfc.MULTIPLE_CHOICE)) && !(item.getTypeId().equals(TypeIfc.MULTIPLE_CORRECT_SINGLE_SELECTION)) && 
+					  !(item.getTypeId().equals(TypeIfc.TRUE_FALSE))) 
+				{
+          return false;
+        }
+      }
+      return true;
+    }
+    return false;
+  }
+
+  public Map getResults() {
+    PublishedAssessmentService pubAssessmentService = new PublishedAssessmentService();
+    Map publishedAnswerHash = pubAssessmentService.preparePublishedAnswerHash(pubAssessmentService.getPublishedAssessment(this.getPublishedId()));
+    GradingService gradingService = new GradingService();
+    Map<String, List<Integer>> resultsByUser = new HashMap<>();
+    for (Object object : agents) {
+      AgentResults agentResults = (AgentResults) object;
+      if (agentResults.getAssessmentGradingId() != -1) {
+        AssessmentGradingData assessmentGradingAux = gradingService.load(agentResults.getAssessmentGradingId().toString());
+        List<Integer> resultsAux = new ArrayList<>(Collections.nCopies(3, 0));
```

Comment:
This entire section needs more documentation about indexes, please document it better.

Reply:
Ok. I put a lot of comments. When I make the pull request confirm if there are too many comments